ground_strike_feature.py

```diff
@@ -0,0 +1,375 @@
+    
+    def compute_ground_strikes(self, df, fighter_a_id, fighter_b_id, index, round_number, last_fights, column_name, differential=False, is_defense=False, is_percentage=False):
+        """
+        Computes the ground strikes for each fighter in the dataset based on column_name
+
+        Parameters:
+        - df (pd.Dataframe): The original dataframe containing all the fights
+        - fighter_a_id (int): The id of the first fighter
+        - fighter_b_id (int): The id of the second fighter
+        - index (int): The index of the current fight
+        - round_number (int): The round number of the current fight
+        - last_fights (int): The number of last fights to consider
+        - column_name (string): The name of the column to compute
+        - differential (bool): Whether to compute the differential or not
+        - is_defense (bool): Whether to compute the defense or not
+        - is_percentage (bool): Whether to compute the percentage or not
+
+        Returns:
+        - (float, float): The ground strikes for each fighter as a tuple
+        """
+        all_prev_fights = df.loc[:index-1]
+        
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id)][-last_fights:]
+            fighter_b_fights = all_prev_fights[(fighter_b_id_vals == fighter_b_id)][-last_fights:]
+
+            sum_a, count_a = self.get_fighter_ground_details(fighter_a_fights, fighter_a_id, round_number, f'total_{column_name}' if round_number == 0 else f'round_{round_number}_{column_name}', is_defense, is_percentage)
+            sum_b, count_b = self.get_fighter_ground_details(fighter_b_fights, fighter_b_id, round_number, f'total_{column_name}' if round_number == 0 else f'round_{round_number}_{column_name}', is_defense, is_percentage)
+
+            return self.get_ground_strike_return_values(sum_a, sum_b, count_a, count_b, differential, is_percentage)
+
+        return 0, 0
+    
+    def get_ground_strike_return_values(self, sum_a, sum_b, count_a, count_b, differential, is_percentage):
+        """
+        Computes the ground strikes averages
+
+        Parameters:
+        - sum_a (float): The sum of ground strikes for fighter a
+        - sum_b (float): The sum of ground strikes for fighter b
+        - count_a (int): The number of fights for fighter a
+        - count_b (int): The number of fights for fighter b
+        - differential (bool): Whether to compute the differential or not
+        - is_percentage (bool): Whether to compute the percentage or not
+
+        Returns:
+        - (float, float): The ground strikes for each fighter as a tuple
+        """
+        fighter_a_ground_strikes = sum_a / count_a if not is_percentage else sum_a
+        fighter_b_ground_strikes = sum_b / count_b if not is_percentage else sum_b
+        
```

Comment:
for the stats like strikes_absorbed is not supposed to be 

> sum(strikes_attempted_op - strikes_attempted_op) / sum(time_fought).  

where we're summing all the strikes_attempted - strikes_landed by op in all the fights then dividing by the total time fought across all fights of interest?

what I think yours is doing is 

> sum(strikes_attempted_op/time_fought) / num_fights

Reply:
Isn't it 

> sum(strikes_landed_op) / sum(tim_fought)

I'm pretty sure absorbed is the amount of strikes that were landed against the fighter

Reply:
ya the strikes absorbed should just be the strikes landed by the opponent, I believe

Reply:
i thought it was the number strikes that didn't land?

Reply:
strikes absorbed is meant to be how strikes did the fighter take or get hit by

Reply:
ok cool, i will update code to do landed/time.